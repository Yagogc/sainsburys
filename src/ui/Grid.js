import styled from "styled-components";

export const GridContainer = styled.main`
  width: 100%;
  max-width: 800px;
  padding: 10px;
  margin: 0 auto;
  text-align: center;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
  align-items: center;
  justify-items: center;
`;

export const GridItem = styled.img`
  width: 100%;
  height: 150px;
  color: red;
  object-fit: cover;
  border: 2px solid ${props => (props.isSelected ? "red" : "#8897a2")}
  transform: scale3D(0.9, 0.9, 1);
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  cursor: pointer;

  &:hover {
    transform: scale3D(1, 1, 1);
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
  }
`;
