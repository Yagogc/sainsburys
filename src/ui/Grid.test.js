import React from "react";
import renderer from "react-test-renderer";
import { GridContainer, GridItem } from "./Grid";

test("UI - GridContainer:", () => {
  const tree = renderer.create(<GridContainer />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("UI - GridItem:", () => {
  const tree = renderer.create(<GridItem />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("border-color", "#8897a2");
  expect(tree).toHaveStyleRule("border-style", "solid");
});

test("UI - GridItem isSelected:", () => {
  const tree = renderer.create(<GridItem isSelected />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("border-color", "red");
  expect(tree).toHaveStyleRule("border-style", "dotted");
});
