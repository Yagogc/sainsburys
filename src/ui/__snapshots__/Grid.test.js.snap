// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI - GridContainer: 1`] = `
.c0 {
  width: 100%;
  max-width: 800px;
  padding: 10px;
  margin: 0 auto;
  text-align: center;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(auto-fill,minmax(150px,1fr));
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

<main
  className="c0"
/>
`;

exports[`UI - GridItem isSelected: 1`] = `
.c0 {
  width: 100%;
  height: 150px;
  color: red;
  object-fit: cover;
  border: 2px;
  border-color: red;
  border-style: dotted;
  -webkit-transform: scale3D(0.9,0.9,1);
  -ms-transform: scale3D(0.9,0.9,1);
  transform: scale3D(0.9,0.9,1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  cursor: pointer;
}

.c0:hover {
  -webkit-transform: scale3D(1,1,1);
  -ms-transform: scale3D(1,1,1);
  transform: scale3D(1,1,1);
  box-shadow: 0 14px 28px rgba(0,0,0,0.25),0 10px 10px rgba(0,0,0,0.22);
}

<img
  className="c0"
/>
`;

exports[`UI - GridItem: 1`] = `
.c0 {
  width: 100%;
  height: 150px;
  color: red;
  object-fit: cover;
  border: 2px;
  border-color: #8897a2;
  border-style: solid;
  -webkit-transform: scale3D(0.9,0.9,1);
  -ms-transform: scale3D(0.9,0.9,1);
  transform: scale3D(0.9,0.9,1);
  box-shadow: 0 1px 3px rgba(0,0,0,0.12),0 1px 2px rgba(0,0,0,0.24);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  cursor: pointer;
}

.c0:hover {
  -webkit-transform: scale3D(1,1,1);
  -ms-transform: scale3D(1,1,1);
  transform: scale3D(1,1,1);
  box-shadow: 0 14px 28px rgba(0,0,0,0.25),0 10px 10px rgba(0,0,0,0.22);
}

<img
  className="c0"
/>
`;
